migration/scripts/migration.js

```diff
@@ -0,0 +1,108 @@
+const serviceAccountUSA = require('./env.production-usa.json');
+const serviceAccountCA = require('./env.production-canada.json');
+
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccountCA),
+});
+const firestore = admin.firestore();
+
+// Reference new Firestore database
+const customersDB = firestore.collection('customers');
+const placesDB = firestore.collection('places');
+const sensorsDB = firestore.collection('sensors');
+const systemsDB = firestore.collection('systems');
+
+// Reference local data (exported from legacy realtime databse)
+const customersData = require('../data/demoCustomer.json');
+const legacyData = require('../data/legacyData-shared.json');
+
+// Categorise data into collections
+const placesData = legacyData.places;
+const sensorsData = legacyData.sensors;
+const systemsData = legacyData.systems;
+
+// Set customers data
+const addCustomers = (id, newDoc) => customersDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+// Object.keys(customersData).forEach((key) => {
+//   addCustomers(key, customersData[key]);
+// });
+
+// Set places data
+const addPlaces = (id, newDoc) => placesDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+Object.keys(placesData).forEach((key) => {
+  // Delete dead data
+  delete placesData[key].admins;
+  delete placesData[key].sensors;
+  placesData[key].customerId = '';
+  addPlaces(key, placesData[key]);
+});
+
+// Set sensors data
+const addSensors = (id, newDoc) => sensorsDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+Object.keys(sensorsData).forEach((key) => {
+  sensorsData[key].customerId = '';
+  // Make systems object an array of systemIds
+  addSensors(key, sensorsData[key]);
+  if (sensorsData[key].systems) {
+    Object.keys(sensorsData[key].systems).forEach((systemId) => {
+      sensorsData[key].systemIds = [systemId];
+    });
+  }
+  // Add datachain as a sub-collection of each sensor doc in sensorsData
+  if (sensorsData[key].datachain) {
+    const datachainRef = firestore.collection('sensors').doc(`${key}`);
+    const { datachain } = sensorsData[key];
+    Object.keys(datachain).forEach((datachainKey) => {
+      const addSubcollection = () => datachainRef
+        .collection('datachain')
+        .doc(datachainKey)
+        .set(datachain[datachainKey])
+        .catch((error) => {
+          console.error(error);
+        });
+      addSubcollection();
+    });
+  }
+  // Exclude legacy datachain & systems structure
+  delete sensorsData[key].datachain;
+  delete sensorsData[key].systems;
+});
+
+// Set systems data
+const addSystems = (id, newDoc) => systemsDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+Object.keys(systemsData).forEach((key) => {
+  addSystems(key, systemsData[key]);
+});
+
+Promise.allSettled([addCustomers, addPlaces, addSensors, addSystems])
```

Comment:
Not sure this is doing anything, this is a list of functions, not iterable list of promises?

these promises are already executed above

This would execute a list of promises and fire when they're done, so instead of `Promise.allSettled([addSystems])`, it would be `Promise.allSettled([addSystems(1), addSystems(2), addSystems(3)])`, not actually like that, but you get the idea that you build out an array of all the promises that are called.